### PR TITLE
Generate UID links for blacklisted user "why"

### PR DESCRIPTION
**Pull request by iBug**

### CODE REVIEW

**Comment by quartata** [comment]
post.post_id is the post that's being scanned, not the post that the user was being blacklisted for

**Comment by quartata** [comment]
the proper fix is to store site, postid instead of the URL really

**Comment by tripleee** [comment]
So is this incorrect now? I like the idea.

**Comment by ArtOfCode-** [comment]
The easy way to do it in this instance would be to just pull the related post ID out of the link - write a regex that takes a link and matches the ID in it.

**Comment by iBug** [comment]
@ArtOfCode- In fact, I'm planning to do it with regex when I have time.

File: helpers.py [hunk 1]
```diff
@@ -103,5 +103,17 @@ def api_parameter_from_link(link):
         return None
 
 
+id_parser_regex = r'https?://[^/]+/[qa]/(\d+)'
```
Comment (angussidney) [inline helpers.py]: I believe we use `/questions/` URLs in a few places, I'm not sure what format we get here.

Comment (iBug) [reply helpers.py]: @angussidney I must have forgot that. Changed it now.

**Pull request merged**